### DISCUSSION TITLE
gfg-potd July 4,2023

### DISCUSSION THREAD

**Discussion by orpheustelos7** [getlost01/gfg-potd#2] [General]
Is this approach based on sliding window algorithm ?

**Comment by getlost01** [comment]
First of all, thank you for reaching me.

Yes, this is the dynamic sliding approach, in which our end pointer `j` iterates and increases by `1` in each iteration and we check that the element product within the starting and ending pointer of our subarray (a kind window) is lesser than `k` or not. If we encounter a condition where the current product of subarray becomes greater than `k`. Then we reduce the size of the subarray (a kind window),  by increasing starting pointer value. 

For more clearance, please refer to the example from the readme file.

**Comment by orpheustelos7** [comment]
Oh, now I get it ,thanks for taking the time to explain me.
